rl/algorithms/learning_algorithms.py

```diff
@@ -179,29 +179,104 @@ def _optimize_model(self):
-            "tau": (ParameterType.FLOAT.name, 0.005, 0, 10),
-            "lr": (ParameterType.FLOAT.name, 1e-4, 0, 10),
-            "use_gpu": (ParameterType.BOOL.name, False, None, None),
-            "seed": (ParameterType.INT.name, 1001, 0, 100000),
-        }
-
-    # Parameters where everything is None should be provided by translator
-    @classmethod
-    def _get_test_params(cls) -> dict:
-        return {
-            "mode": (ParameterType.STRING.name, None, None, None),
-            "use_gpu": (ParameterType.BOOL.name, True, None, None),
+            "n_observations": Parameter(
+                ParameterType.INT.name,
+                2720,
+                0,
+                None,
+                "Number of observations in the state",
+                False,
+            ),
+            "n_actions": Parameter(
+                ParameterType.INT.name,
+                376,
+                0,
+                None,
+                "Number of actions in the state",
+                False,
+            ),
+            "eps_start": Parameter(
+                ParameterType.FLOAT.name,
+                0.9,
+                0,
+                1,
+                "Probability of choosing random action at the beginning",
+                True,
+            ),
+            "eps_end": Parameter(
+                ParameterType.FLOAT.name,
+                0.05,
+                0,
+                1,
+                "Probability of choosing random action at the end",
+                True,
+            ),
+            "eps_decay": Parameter(
+                ParameterType.FLOAT.name,
+                1000,
+                0,
+                None,
+                "Number of steps over which eps is linearly annealed",
+                True,
+            ),
+            "memory_size": Parameter(
+                ParameterType.INT.name,
+                10000,
+                1,
+                None,
+                "Number of transitions stored in memory",
+                True,
+            ),
+            "batch_size": Parameter(
+                ParameterType.INT.name,
+                128,
+                1,
+                2048,
+                "Number of transitions used for training in one batch",
+                True,
+            ),
+            "gamma": Parameter(
+                ParameterType.FLOAT.name,
+                0.99,
+                0,
+                1,
+                "Discount factor for future rewards",
+                True,
+            ),
+            "tau": Parameter(
+                ParameterType.FLOAT.name,
+                0.005,
+                0,
+                1,
+                "Soft update of target network's weights",
+                True,
+            ),
+            "lr": Parameter(
+                ParameterType.FLOAT.name,
+                1e-4,
+                0,
+                None,
+                "Learning rate for Adam optimizer",
+                True,

```

Comment:
To poruszałem z tobą na priv, więc tutaj nie będę się zbytnio rozpisywał.

Nie może być sytuacji w której jeden parametr po modyfikacji od razu działa bez zarzutów, a drugi wymaga wczytania od nowa modelu.

Podczas metody update_config trzeba pomyśleć jak to zrobić tak, żeby zmiana dowolnego parametru "modyfikowalnego" od razu wchodziła w życie.

Reply:
to będzie trzeba dodać w innym pull requeście, to ze obecnie algorytmy nie obsługują takich zmian to nie bug, tylko brakującu feature.